develop

leanbre opened the pull request.

loana13 committed:
feat: тесты на страницу с основными вопросами
feat: + заготовка для перехода по кнопкам "Заказать"

loana13 committed:
feat: добавлен объект страницы заказа
feat: добавлены тесты на happy path заказа самоката на обе точки входа в сценарий

loana13 committed:
feat: добавлен объект страницы заказа
feat: добавлены тесты на happy path заказа самоката на обе точки входа в сценарий

okryukova commented on pom.xml:
⚠️Можно улучшить. Хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройку (если необходима) и как запускать (mvn clean test - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблицы используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

okryukova commented on src/main/java/model/HomePage.java:
```diff
@@ -0,0 +1,71 @@
+    public static final String IMPORTANT_ANSWER_PREFIX = "accordion__panel-";
+
+    // Локатор секции "Вопросы о важном"
+    public static final By importantQuestionsSectionLocator = By.xpath(".//div[text()='Вопросы о важном']");
```
⚠️Можно улучшить. В классическом pojo поля класса приватные.
⛔️Нужно исправить.  Имена констант должны состоять из заглавных букв. Например, SNAKE_UPPER_CASE.

okryukova commented on src/main/java/model/OrderPage.java:
```diff
@@ -0,0 +1,218 @@
+    private final WebDriver driver;
+
+    // Базовая ссылка на основную страницу
+    public static final String BASE_URL = "https://qa-scooter.praktikum-services.ru/";
```
⛔️Нужно исправить.  Константа задана уже в HomePage

okryukova commented on src/main/java/model/OrderPage.java:
```diff
@@ -0,0 +1,218 @@
+    public static final String SUCCESS_PLACED_ORDER_TEXT = "Заказ оформлен";
+
+    // Локатор кнопки скрытия информации о куках
+    private final By cookieButton = By.id("rcc-confirm-button");
```
⛔️Нужно исправить.  Куки и кнопки заказать относятся к HomePage, перенеси их, пожалуйста

okryukova commented on src/test/java/ImporantQuestionsTest.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    @Parameterized.Parameters()
+    public static Object[][] questionsTexts() {
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

okryukova commented on src/test/java/ImporantQuestionsTest.java:
```diff
@@ -0,0 +1,65 @@
+    public static Object[][] questionsTexts() {
+
+        return new Object[][]{
+                {0, "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
```
⚠️Можно улучшить. Тестовые данные  должны содержаться либо в константах тестового класса, либо в отдельном файле

okryukova commented on src/test/java/ImporantQuestionsTest.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    @After
+    public void teardown() {
```
⚠️Можно улучшить. Блоки закрытия и открытия  браузера одинаковы для всех тестовых классов. Целесообразно вынести их в базовый или утилитный класс.

okryukova commented on src/test/java/ImporantQuestionsTest.java:
```diff
@@ -0,0 +1,65 @@
+    @Before
+    public void beforeTest() {
+        WebDriverManager.chromedriver().setup();
+        driver = new ChromeDriver();
```
⚠️Можно улучшить. Браузер, как правило, запускают в полноэкранном режиме (maximized) или определенном размере окна.

okryukova commented on src/test/java/ImporantQuestionsTest.java:
```diff
@@ -0,0 +1,65 @@
+        };
+    }
+    @Test
+    public void checkImportantQuestionsHaveCorrectAnswers() {
```
⚠️Можно улучшить. В названии метода необходимо использовать слово Test. Проверь, соблюдается ли данное правило для всех созданных тестов.

okryukova commented on src/test/java/OrderFlowTest.java:
```diff
@@ -0,0 +1,137 @@
+        // Заполняем данные на первом экране
+        orderPage.setName(name);
+        orderPage.setSurname(surname);
+        orderPage.setDeliveryAddress(deliveryAddress);
```
⚠️Можно улучшить. В теме UI-тестирование > Page Object > Работа с методами > Объединение методов хорошо описан прием объединения методов. Применяй его для сокращения кода автотеста, где это необходимо. Например, при заполнении форм.

loana13 committed:
fix: поля классов OrderPage и HomePage сделаны приватными и переведены в upper snake case
fix: убрана константа URL из OrderPage, заиспользована из HomePage
fix: кнопки "заказать" и кнопка скрытия кук перенесены в HomePage (локатор и действие нажатия)
